code/game/machinery/mind_machine.dm

```diff
@@ -0,0 +1,856 @@
+	anchored = 1
+	use_power = 1
+	idle_power_usage = 50
+	active_power_usage = 2000
+	light_power_on = 1
+	machine_flags = SCREWTOGGLE | CROWDESTROY | EMAGGABLE | FIXED2WORK | WRENCHMOVE
+
+#define CURRENTLY_ACTIVE "Currently active: please wait"
+#define PODS_UNLOCKED "Pods already unlocked"
+#define PODS_LOCKED "Pods are locked"
+#define REQ_SCAN "Require scan"
+#define DELAY_CAUTION "Proceed with caution: supervisor advised"
+#define MISSING_OCC "Missing occupant"
+#define CONDUIT_CHARGE "Conduit requires charge"
+#define SHIELD_ERROR "Critical error, aborting"
+#define LIVING_REQUIRED "Living mind required"
+#define SHARD_REACTION "Warning: Unknown internal reaction"
+#define SCANNER_INSUFF "Scanners insufficient"
+#define UNKNOWN_INT "Unknown interruption"
+#define SUM_TOO_LOW "Sum intelligence too low"
+#define DELIVERY_ERROR "Error in mind delivery"
+#define MIND_DAMAGED "Mind sustained damage"
+#define SAFETY_CONSUMED "Safety consumed"
+#define TRANSPORT_ANOM "Mind transport anomaly"
+#define NOERROR "No errors"
+
+#define LOWER "Lower"
+#define HIGHER "Higher"
+#define SILICON "Artificial"
+#define SHIELDED "Shielded"
+
+/obj/machinery/mind_machine/mind_machine_hub
+	name = "\improper Mind Machine Hub"
+	icon_state = "mind_hub"
+	desc = "The main hub of a complete mind machine setup. Placed between two mind pods and used to control and manage the transfer. Houses an experimental bluespace conduit which uses bluespace crystals for charge."
+	var/bluespaceConduit = 0 //Swap ammo
+	var/soulShardSafety = FALSE	//Day ruin insurance
+	var/obj/machinery/mind_machine/mind_machine_pod/connectOne //All "variableNameOne" refer to information about the connectOne pod
+	var/obj/machinery/mind_machine/mind_machine_pod/connectTwo //Likewise for variableNameTwo for connectTwo
+	var/podsConnected = FALSE
+	var/mob/living/occupantOne = null
+	var/mob/living/occupantTwo = null
+	var/occupantStatOne = null
+	var/occupantStatTwo = null
+	var/mindTypeOne = null	//Player mind, simple mob, silicon, etc
+	var/mindTypeTwo = null
+	var/lockedPods = FALSE
+	var/currentlySwapping = FALSE
+	var/swapProgress = 0
+	var/list/theFly = list() //If more than one creature enters a pod, ie: human with borer carrying a carp in his bag
+	var/occupantScan = FALSE
+	var/manipRating = 4 //Determines speed of swap/malfunction chance on delay swap
+	var/scanRatingOne = 6 //Determines if you can swap silicons
+	var/scanRatingTwo = 6
+	var/errorMessage = NOERROR
+	var/beenSwapped = FALSE
+	var/malfSwap = FALSE //Triggers malfunction if true
+	var/list/illegalSwap = list() //Might break the game/remove the player from the round. If you get mechahitler in there you deserve your round, though
+
+//////Parts and connection/////////
+
+/obj/machinery/mind_machine/mind_machine_hub/New()
+	illegalSwap = boss_mobs + blacklisted_mobs - list(/mob/living/simple_animal/hostile/mechahitler, /mob/living/simple_animal/hostile/alien/queen/large, /mob/living/simple_animal/hostile/retaliate/cockatrice)
```

Comment:
>cockatrice

*[foreseeable consequences intensifies]*

Reply:
I foresee fun
Fun and statues